### PR TITLE
feat: add writing managed accounts feature in terraform provider

### PR DESCRIPTION
**Pull request by btfhernandez**
Changes from go client library side.

### CODE REVIEW

File: api/managed_account/managed_account.go [hunk 1]
```diff
@@ -250,3 +249,144 @@ func (managedAccounObj *ManagedAccountstObj) ManagedAccountRequestCheckIn(reques
...
+	var managedSystem *entities.ManagedSystemResponse
+	var createResponse entities.CreateManagedAccountsResponse
+
+	accountDetails, err := utils.ValidateCreateManagedAccountInput(accountDetails)
+
+	if err != nil {
+		return createResponse, err
+	}
+
+	ManagedAccountSytemUrl := managedAccounObj.authenticationObj.ApiUrl.JoinPath("ManagedSystems").String()
```
Comment (dtejadav) [inline api/managed_account/managed_account.go:265]: Typo "Sytem"

File: api/managed_account/managed_account.go [hunk 2]
```diff
@@ -250,3 +249,144 @@ func (managedAccounObj *ManagedAccountstObj) ManagedAccountRequestCheckIn(reques
...
+			managedSystem = &v
+			break
+		}
+	}
+
+	if managedSystem == nil {
+		return createResponse, fmt.Errorf("managed system %v was not found in managed system list", systemNameTarget)
+	}
+
+	ManagedAccountCreateManagedAccountUrl := managedAccounObj.authenticationObj.ApiUrl.JoinPath("ManagedSystems", fmt.Sprintf("%d", managedSystem.ManagedSystemID), "ManagedAccounts").String()
```
Comment (dtejadav) [inline api/managed_account/managed_account.go:283]: Typo "Accoun"

File: api/managed_account/managed_account.go [hunk 3]
```diff
@@ -250,3 +249,144 @@ func (managedAccounObj *ManagedAccountstObj) ManagedAccountRequestCheckIn(reques
...
+	if err != nil {
+		return createResponse, err
+	}
+
+	return createResponse, nil
+
+}
+
+// ManagedAccountCreateManagedAccount calls Secret Safe API Requests enpoint to create managed accounts.
+func (managedAccounObj *ManagedAccountstObj) ManagedAccountCreateManagedAccount(accountDetails entities.AccountDetails, url string) (entities.CreateManagedAccountsResponse, error) {
```
Comment (dtejadav) [inline api/managed_account/managed_account.go:295]: Typo "Accoun"

File: api/managed_account/managed_account.go [hunk 4]
```diff
@@ -250,3 +249,144 @@ func (managedAccounObj *ManagedAccountstObj) ManagedAccountRequestCheckIn(reques
...
+	accountDetailsJsonString := string(accountDetailsJson)
+
+	b := bytes.NewBufferString(accountDetailsJsonString)
+
+	var body io.ReadCloser
+	var technicalError error
+	var businessError error
+
+	technicalError = backoff.Retry(func() error {
+		body, _, technicalError, businessError = managedAccounObj.authenticationObj.HttpClient.CallSecretSafeAPI(url, "POST", *b, "ManagedAccountCreateManagedAccount", "", "")
```
Comment (dtejadav) [inline api/managed_account/managed_account.go:313]: Typo "Accoun"

**Review by dtejadav — CHANGES_REQUESTED** [review]

File: api/managed_account/managed_account.go [hunk 5]
```diff
@@ -250,3 +249,144 @@ func (managedAccounObj *ManagedAccountstObj) ManagedAccountRequestCheckIn(reques
...
+	var createResponse entities.CreateManagedAccountsResponse
+
+	accountDetails, err := utils.ValidateCreateManagedAccountInput(accountDetails)
+
+	if err != nil {
+		return createResponse, err
+	}
+
+	ManagedAccountSytemUrl := managedAccounObj.authenticationObj.ApiUrl.JoinPath("ManagedSystems").String()
+	managedSystemGetSystemsResponse, err := managedAccounObj.ManagedSystemGetSystems(ManagedAccountSytemUrl)
```
Comment (RanjitMishraBT) [inline api/managed_account/managed_account.go:266]: typo "Accoun"

**Review by dtejadav — APPROVED** [review]
Nice work!

**Pull request merged**